src/aap_eda/core/models/rulebook.py

```diff
@@ -36,11 +37,26 @@ class Meta:
     name = models.TextField(null=False)
     description = models.TextField(null=True, default="")
     # TODO: this field should not have a default value.
+    # TODO: should the content of this field be validated?
+    # https://issues.redhat.com/browse/AAP-19202
     rulesets = models.TextField(null=False, default="")
     project = models.ForeignKey("Project", on_delete=models.CASCADE, null=True)
     created_at = models.DateTimeField(auto_now_add=True, null=False)
     modified_at = models.DateTimeField(auto_now=True, null=False)
 
```

Comment:
@Alex-Izquierdo Do we need a attribute in the Rulebook model that has a boolean field that stores token_required.
token_required = models.BooleanField()

Which can be set once we parse the rulebook.

The UI would need this when it detects that a rulebook needs a token, it can prompt the user to pick a token or if the rulebook doesn't need the token it can hide the token selection.

Reply:
I'm not aware of such a requirement. I understood that the check would be reactive. I agree it can improve the UX but this would add more effort in the implementation for the UI and for the API (where we have to change as well the import/sync logic and the migrations). 

Let's discuss it with @eclarizio and @lgalis to be everybody aligned on this or even consider to implement it in later iterations.